server/app/org/scalaexercises/exercises/services/ExercisesService.scala

```diff
@@ -26,12 +28,9 @@ object ExercisesService extends RuntimeSharedConversions {
   lazy val (errors, runtimeLibraries) = Exercises.discoverLibraries(cl = classLoader)
 
   lazy val (libraries: List[Library], librarySections: Map[String, List[Section]]) = {
-    val libraries1 = colorize(runtimeLibraries)
+    val libraries1 = colorize(runtimeLibraries).map(convertLibrary)
     errors.foreach(error ⇒ Logger.warn(s"$error")) // TODO: handle errors better?
-    (
-      libraries1.map(convertLibrary),
-      libraries1.map(library0 ⇒ library0.name → library0.sections.map(convertSection)).toMap
-    )
+    (libraries1, libraries1.map(lib ⇒ lib.name → lib.sections).toMap)
```

Comment:
I think `convertSection` was called twice for every section, once in `convertLibrary` and once explicitly when creating the `Section` map).